### PR TITLE
DOC-1671: Commit generated API docs

### PR DESCRIPTION
**Pull request by MitchCTiny**
Related Ticket: DOC-1671

Description of Changes:
* Generated API docs
* Removed API docs from .gitignore
* Updated README.md with instructions to generate API docs

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] `modules/ROOT/moxiedoc_nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- ~[x] Files removed have been deleted, not just excluded from the build (if applicable)~
- ~[x] (New product features only) Release Note added~

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ The `yarn build` step will download the latest TinyMCE package and generate new
 
 #### TinyMCE API documentation
 
-The TinyMCE API documentation is maintained within the [TinyMCE project repository](https://github.com/tinymce/tinymce) and compiled for the documentation site using [MoxieDoc](https://github.com/tinymce/moxiedoc).
+The TinyMCE API documentation is maintained within the [TinyMCE project repository](https://github.com/tinymce/tinymce) and compiled for the documentation site using the above `yarn build` command.
```
Comment (lnewson) [inline README.md:71]: This change is incorrect... it's still compiled by MoxieDoc.

- [ ] Revert this change please

**Review by lnewson — CHANGES_REQUESTED** [review]
Looks like part of the DoD hasn't been done here:
> The README or CONTRIBUTING docs have been updated to reference how to update the API docs.

This hasn't been done and we need a new section added to specify how to update the API docs, especially since the process is changing from it being able to automatically build.

File: README.md (hunk 1 above)
Comment (MitchCTiny) [reply README.md]: Reverted

File: README.md [hunk 2]
```diff
@@ -70,6 +70,8 @@ The `yarn build` step will download the latest TinyMCE package and generate new
 
 The TinyMCE API documentation is maintained within the [TinyMCE project repository](https://github.com/tinymce/tinymce) and compiled for the documentation site using [MoxieDoc](https://github.com/tinymce/moxiedoc).
 
+Running `yarn build` will download the latest TinyMCE package and generate new API reference content from source code. This API documentation must be generated and updated on the documentation site with each release.
```
Comment (lnewson) [inline README.md:73]:
```suggestion
To update the published API documentation, change the version in the `.api-version` file, run `yarn build` and then commit the changes. Running `yarn build` will download the TinyMCE package specified in `.api-version` and generate new API reference content from source code.

**Note:** The API documentation should never be edited manually.
```
We don't need the last sentence and it doesn't download the latest version so that part isn't quite correct. Also we're missing the details about committing the changes to make it appear, so I think the above should address those issues.

Comment (MitchCTiny) [reply README.md:73]: Thanks @lnewson

**Review by lnewson — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -70,6 +70,10 @@ The `yarn build` step will download the latest TinyMCE package and generate new
 
 The TinyMCE API documentation is maintained within the [TinyMCE project repository](https://github.com/tinymce/tinymce) and compiled for the documentation site using [MoxieDoc](https://github.com/tinymce/moxiedoc).
 
+To update the published API documentation, change the version in the `.api-version` file, run `yarn build` and then commit the changes. Running `yarn build` will download the TinyMCE package specified in `.api-version` and generate new API reference content from source code.
```
Comment (brian-forte) [inline README.md:73]:
```suggestion
To update the published API documentation:

1. change the version in `.api-version`.
2. run `yarn build`.
3. commit the changes.

Running `yarn build` downloads the TinyMCE package specified in `.api-version` and generates new API reference content from source.
```
Putting procedures in lists makes them easier to follow (and makes folks slightly less likely to skip a step).

Also some minor edits.

File: README.md [hunk 4]
```diff
@@ -70,6 +70,10 @@ The `yarn build` step will download the latest TinyMCE package and generate new
...
+
+**Note:** The API documentation should never be edited manually.
```
Comment (brian-forte) [inline README.md:75]:
```suggestion
NOTE: The API documentation should never be edited manually.
```

*NOTE:* is an [admonition block](https://docs.asciidoctor.org/asciidoc/latest/blocks/admonitions/). It doesn’t need to be marked-up for presentation purposes, our CSS will do that for us.

File: modules/ROOT/pages/apis/tinymce.addonmanager.adoc [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+[[methods]]
+== Methods
+
+[[add]]
+=== add()
+[source, javascript]
+----
+add(id: String, addOn: tinymce.Theme | tinymce.Plugin): tinymce.Theme | tinymce.Plugin
+----
+Adds a instance of the add-on by it's short name.
```
Comment (brian-forte) [inline modules/ROOT/pages/apis/tinymce.addonmanager.adoc:32]:
```suggestion
Adds an instance of the add-on by its short name.
```
Typo corrections.

1. The indefinite article is *an* when the following sound is a vowel.
2. *Its* is the possessive pronoun. *It’s* is the abbreviation. FWIW, I remember this ‘exception’ by remembering that no possessive pronouns take an apostrophe (so: his, hers, theirs, its).

File: modules/ROOT/pages/apis/tinymce.addonmanager.adoc [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+----
+
+==== Parameters
+
+* `name (String)` - Short name of the add-on that gets loaded.
+* `addOnUrl (String)` - URL to the add-on that will get loaded.
+
+==== Return value
+
+* `Promise` - A promise that will resolve when the add-on is loaded successfully or reject if it failed to load.
```
Comment (brian-forte) [inline modules/ROOT/pages/apis/tinymce.addonmanager.adoc:113]:
```suggestion
* `Promise` — A promise that will resolve when the add-on loads successfully or reject if the addon fails to load.
```

1. Replaced the hyphen set open with an em-dash (also set open). The dash here is being used as a linking mark (it connects the word with its definition) and the em-dash is the grammatically accurate dash to use in this instance.
2. Edited the definition for tense. The original began in the simple present, switched to the simple past, then switched to the present participle. The edit is all in the simple present.

File: modules/ROOT/pages/apis/tinymce.annotator.adoc [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+[[methods]]
+== Methods
+
+[[annotate]]
+=== annotate()
+[source, javascript]
+----
+annotate(name: String, data: Object)
+----
+Applies the annotation at the current selection using data
```
Comment (brian-forte) [inline modules/ROOT/pages/apis/tinymce.annotator.adoc:35]:
```suggestion
Applies the annotation at the current selection using data.
```
Missing full-stop.

File: modules/ROOT/pages/apis/tinymce.annotator.adoc [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+==== Parameters
+
+* `name (String)` - the name of the annotation to apply
```
Comment (brian-forte) [inline modules/ROOT/pages/apis/tinymce.annotator.adoc:39]:
```suggestion
* `name (String)` — the name of the annotation to apply.
```
Open hyphen replaced with open em-dash. Missing full-stop no longer missing.

File: modules/ROOT/pages/apis/tinymce.annotator.adoc [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+* `data (Object)` - information to pass through to this particular
```
Comment (brian-forte) [inline modules/ROOT/pages/apis/tinymce.annotator.adoc:40]:
```suggestion
* `data (Object)` — information to pass through to this particular.
```
Open hyphen replaced with open em-dash. Missing full-stop no longer missing.

File: README.md (hunk 4 above)
Comment (lnewson) [reply README.md:75]: @brian-forte this is **Markdown** not AsciiDoc 😉

File: modules/ROOT/pages/apis/tinymce.addonmanager.adoc (hunk 5 above)
Comment (lnewson) [reply modules/ROOT/pages/apis/tinymce.addonmanager.adoc:32]: @brian-forte I think you missed an important part of the README updates here:
> **Note:** The API documentation should never be edited manually.

So as noted, this cannot be changed like this and any of these changes would need to be done on the TinyMCE source.

Comment (lnewson) [reply modules/ROOT/pages/apis/tinymce.addonmanager.adoc:32]: I'd suggest logging these changes in a Jira and getting them fixed that way instead.

**Review by brian-forte — APPROVED** [review]
